reconcile/utils/terrascript_client.py

```diff
@@ -4085,7 +4130,71 @@ def populate_tf_resource_secrets_manager(self, resource, namespace_info):
         for tf_resource in tf_resources:
             self.add_resource(account, tf_resource)
 
-    def populate_tf_resource_asg(self, resource, namespace_info):
+    def get_asg_image_id(self, image, existing_secrets, output_prefix,
```

Comment:
i think you can simplify things a bit here...
if the "last known commit" (existing secrets) matches what is currently in the repo - we are done.
else, if there is a new commit, check if the image exists and shared, etc. if any of these conditions is not met - use the "last known commit".